packages/core/src/lib/engine.core.ts

```diff
@@ -262,38 +262,113 @@ export class DiagramEngineCore {
   }
 
   /**
-   * fit the canvas zoom levels to the elements contained.
-   * @param additionalZoomFactor allow for further zooming out to make sure edges doesn't cut
+   * Get the bounding rectangle of the input group of nodes
+   * @param nodes the group of nodes to calculate the retcangle
+   * @param margin the desired margin to include when calc the rect (in px)
+   * @returns The total width and height of the nodes, most top and most left points of the nodes group
    */
-  zoomToFit(additionalZoomFactor = 0.005) {
-    this.canvas$.value$
-      .pipe(
-        filter(
-          (canvas: HTMLElement | null | undefined): canvas is HTMLElement =>
-            canvas !== null && canvas !== undefined
-        ),
-        take(1),
-        delay(0)
-      )
-      .subscribe((canvas) => {
-        const diagramModel = this.getDiagramModel();
-        const xFactor = canvas.clientWidth / canvas.scrollWidth;
-        const yFactor = canvas.clientHeight / canvas.scrollHeight;
-        const zoomFactor = xFactor < yFactor ? xFactor : yFactor;
-
-        let newZoomLvl =
-          diagramModel.getZoomLevel() * (zoomFactor - additionalZoomFactor);
-        const maxZoomOut = diagramModel.getMaxZoomOut();
-
-        if (maxZoomOut && newZoomLvl < maxZoomOut) {
-          newZoomLvl = maxZoomOut;
-        }
+  getNodeLayersRect(nodes: NodeModel[], margin = 0) {
+    if (!nodes || nodes.length === 0) {
+      return { width: 0, height: 0, top: 0, left: 0 };
```

Comment:
throwing here might be more suitable since this functions are exposed to the library consumers